### DISCUSSION TITLE
Files with identical titles not appearing in Docs menu

### DISCUSSION THREAD

**Discussion by edornshuld** [thuliteio/doks#117] [Q&A]
If I have

```
├── archive/
│   ├── 2017/
│   │   ├── information.md
│   ├── 2016/
│   │   ├── information.md
│   └── _index.md
```

where the nested `.md` files have a title that reflects their filename (i.e. `title: "Information"` for `information.md`). Each `.md` file has its own unique parent defined (e.g. `parent: "2017"` and `parent: "2016"`).

Unfortunately, the docs menu renders as

**2017**
**2016**
Information

I want it to show

**2017**
Information
**2016**
Information

If I give the `.md` files unique titles such as

`title: "2016 Information"` for `archive/2016/information.md` and
`title: "2017 Information"` for `archive/2017/information.md` 

the docs menu shows

**2016**
2016 Information
**2017**
2017 Information

Is there a way to have files with identical filenames that are recognized by the subfolder they are in and shown under their respective `parent` in the docs menu?  Can the menu be sorted in reverse chronological order?

**Comment by epogrebnyak** [comment]
Try looking at weights in markdown frontmatter, write back if no solution

**Comment by edornshuld** [comment]
```
├── archive/
│   ├── 2017/
│   │   ├── information.md
│   ├── 2016/
│   │   ├── information.md
│   └── _index.md
```

If I give `2017/information.md` a weight of 10 and `2016/information.md` a weight of 20, my menu looks like this

**2016**
Information

The 2017 parent title does not appear in the menu. Is there something I'm missing (somewhere else I should be looking)?

EDIT: If I remove `2016/information.md`, 2017 will appear properly for `2017/information.md` in the docs menu.


**2016**
**2017**
Information

**Comment by h-enk** [comment]
> Is there a way to have files with identical filenames that are recognized by the subfolder they are in and shown under their respective `parent` in the docs menu? Can the menu be sorted in reverse chronological order?

Yes there is. See section [Menu Entry Variables](https://gohugo.io/variables/menus/#menu-entry-variables) from the Hugo docs:

> It is necessary to set a unique identifier manually if two or more menu entries have the same `.Name`.

So, add unique identifiers like so e.g.:

```md
---
title: "Information"
description: ""
lead: ""
date: 2021-02-07T15:00:02+01:00
lastmod: 2021-02-07T15:00:02+01:00
draft: false
images: []
menu:
  docs:
    parent: "2017"
    identifier: "info-2017"
weight: 200
toc: true
---
```

```md
---
title: "Information"
description: ""
lead: ""
date: 2021-02-07T15:00:07+01:00
lastmod: 2021-02-07T15:00:07+01:00
draft: false
images: []
menu:
  docs:
    parent: "2016"
    identifier: "info-2016"
weight: 210
toc: true
---
```

And, make sure to add to `./config/_default/menus.toml`:

```toml

[[docs]]
  name = "2017"
  weight = 15
  identifier = "2017"
  url = "/docs/archive/2017/"

[[docs]]
  name = "2016"
  weight = 20
  identifier = "2016"
  url = "/docs/archive/2016/"
```

Like @epogrebnyak indicated, you can use `weight` to set your preferred order.

The result:

![2021-02-07_15-39-17](https://user-images.githubusercontent.com/3902872/107149746-b46dcd00-695a-11eb-8408-4b114c58edbc.png)

**Reply by edornshuld** [reply]
Perfect. This worked. Thank you and thanks for the Doks development.

**Reply by ImaCrea** [reply]
thanks! working :) was having quiet a headache on this one. beautiful community here!

**Comment by edornshuld** [comment]
UPDATE QUESTION:

Menu is working great.  However, when I click on one of the pages, all the other similarly named pages get highlighted.

Example menu:

**2018**
Information
**2017**
Information

When I click on _either_ "Information" link in the doks menu, BOTH get highlighted purple.

Anyway we can limit the color to the selected menu option?

**Reply by h-enk** [reply]
Hey @edornshuld, nice one 😉 

The active class is matched on the title of the current page — matching on identifier is what we want.

In `./layouts/partials/sidebar/docs-menu.html` replace line 8: 

```html
{{- $active = or $active (eq .Name $currentPage.Title) -}}
``` 

with:

```html
{{- $active = or $active (eq $currentPage.Section .Identifier) -}}
```

I will update Doks accordingly.

**Reply by edornshuld** [reply]
Excellent. Thank you!
